multi-node/aws/pkg/config/config.go

```diff
@@ -203,6 +228,25 @@ func (c Cluster) stackConfig(opts StackTemplateOptions, compressUserData bool) (
 
 	stackConfig.Config.TLSConfig = compactAssets
 
+	controllerIPAddr := net.ParseIP(stackConfig.ControllerIP)
+	if controllerIPAddr == nil {
+		return nil, fmt.Errorf("invalid controllerIP: %s", stackConfig.ControllerIP)
+	}
+	controllerSubnetFound := false
+	lastSubnetIndex := len(stackConfig.Subnets) - 1
+	for i, subnet := range stackConfig.Subnets {
+		_, instanceCIDR, err := net.ParseCIDR(subnet.InstanceCIDR)
+		if err != nil {
+			return nil, fmt.Errorf("invalid instanceCIDR: %v", err)
+		}
+		if instanceCIDR.Contains(controllerIPAddr) {
+			stackConfig.ControllerSubnetIndex = i
+			controllerSubnetFound = true
+		} else if !controllerSubnetFound && i == lastSubnetIndex {
+			return nil, fmt.Errorf("Fail-fast occurred possibly because of a bug: ControllerSubnetIndex couldn't be determined for subnets (%v) and controllerIP (%v)", stackConfig.Subnets, stackConfig.ControllerIP)
+		}
```

Comment:
If I understand this loop correctly, I think it could be made a bit more simple by dropiing `lastSubnetIndex` and moving this final check
outside of the loop:

```
	controllerSubnetFound := false
	for i, subnet := range stackConfig.Subnets {
		_, instanceCIDR, err := net.ParseCIDR(subnet.InstanceCIDR)
		if err != nil {
			return nil, fmt.Errorf("invalid instanceCIDR: %v", err)
		}
		if instanceCIDR.Contains(controllerIPAddr) {
			stackConfig.ControllerSubnetIndex = i
			controllerSubnetFound = true
		}
	}
	if !controllerSubnetFound {
		return nil, fmt.Errorf("Fail-fast occurred possibly because of a bug: ControllerSubnetIndex couldn't be determined for subnets (%v) and controllerIP (%v)", stackConfig.Subnets, stackConfig.ControllerIP)
	}
```